Trimble.Modus.Components/Controls/TMCheckBox/TMCheckBox.cs

```diff
@@ -0,0 +1,188 @@
+﻿using Microsoft.Maui.Controls;
+using System.Drawing;
+using Trimble.Modus.Components.Enums;
+
+namespace Trimble.Modus.Components
+{
+    public class TMCheckBox : ContentView
+    {
+        private Label _label;
+        private Image _checkbox;
+        private int _default_width = 24, _default_height = 24, _large_width = 32,  _large_height = 32;
+        private int _default_font_size = 14, _large_font_size = 16;
+
+
+        public bool IsDisabled
+        {
+            get => (bool)GetValue(IsDisabledProperty);
+            set => SetValue(IsDisabledProperty, value);
+        }
+        public CheckboxSize Size
+        {
+            get => (CheckboxSize)GetValue(SizeProperty);
+            set => SetValue(SizeProperty, value);
+        }
+        public string Text
+        {
+            get => (string)GetValue(TextProperty);
+            set => SetValue(TextProperty, value);
+        }
+        public bool IsChecked
+        {
+            get => (bool)GetValue(IsCheckedProperty);
+            set => SetValue(IsCheckedProperty, value);
+        }
+        public bool IsIndeterminate
+        {
+            get => (bool)GetValue(IsIndeterminateProperty);
+            set => SetValue(IsIndeterminateProperty, value);
+        }
+
+        public static readonly BindableProperty IsDisabledProperty =
+         BindableProperty.Create(
+             nameof(IsDisabled),
+             typeof(bool),
+             typeof(TMCheckBox),
+             true,
+             propertyChanged: OnIsDisabledChanged);
+
+        public static readonly BindableProperty IsIndeterminateProperty =
+          BindableProperty.Create(
+              nameof(IsIndeterminate),
+              typeof(bool),
+              typeof(TMCheckBox),
+              false,
+              propertyChanged: OnIsIndeterminateChanged);
+
+        private static void OnIsIndeterminateChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            var customCheckboxView = (TMCheckBox)bindable;
+
+            if (customCheckboxView.IsIndeterminate) { 
+                customCheckboxView.IsChecked = false;
```

Comment:
Why both case isChecked=false

Reply:
Removed Unwanted statement of setting state